lib/irc/ClientPool.js

```diff
@@ -4,11 +4,12 @@
  * and may be closed for a variety of reasons.
  */
 "use strict";
-var stats = require("../config/stats");
-var log = require("../logging").get("ClientPool");
-var Promise = require("bluebird");
+const stats = require("../config/stats");
+const log = require("../logging").get("ClientPool");
+const Promise = require("bluebird");
+const QueuePool = require("../util/QueuePool");
 
-const RECONNECT_TIME_MS = 1000;
+const CONCURRENT_RECONNECT_LIMIT = 50;
```

Comment:
You almost certainly want this to be configurable in the YAML file, alongside `reconnectIntervalMs`.